notebooks/examples/python/Ship2ShipTransfers/00. Data Prep.py

```diff
@@ -0,0 +1,200 @@
+mos.enable_mosaic(spark, dbutils)
+
+# COMMAND ----------
+
+# MAGIC %md ##AIS Data
+
+# COMMAND ----------
+
+dbutils.fs.mkdirs("/tmp/vessels")
+
+# COMMAND ----------
+
+# MAGIC %sh
+# MAGIC # see: https://coast.noaa.gov/htdata/CMSP/AISDataHandler/2018/index.html
+# MAGIC # we download data to dbfs:// mountpoint (/dbfs)
+# MAGIC cd /dbfs/tmp/vessels/
+# MAGIC
+# MAGIC wget -np -r -nH -L --cut-dirs=3 https://coast.noaa.gov/htdata/CMSP/AISDataHandler/2018/AIS_2018_01_31.zip > /dev/null 2>&1
+# MAGIC unzip 2018/AIS_2018_01_31.zip
+
+# COMMAND ----------
+
+schema = """
+  MMSI int, 
+  BaseDateTime timestamp, 
+  LAT double, 
+  LON double, 
+  SOG double, 
+  COG double, 
+  Heading double, 
+  VesselName string, 
+  IMO string, 
+  CallSign string, 
+  VesselType int, 
+  Status int, 
+  Length int, 
+  Width int, 
+  Draft double, 
+  Cargo int, 
+  TranscieverClass string
+"""
+
+AIS_df = (
+    spark.read.option("badRecordsPath", "/tmp/ais_invalid")  # Quarantine bad records
+    .csv("/tmp/vessels/2018", header=True, schema=schema)
+    .filter("VesselType = 70")  # Only select cargos
+    .filter("Status IS NOT NULL")
+)
+display(AIS_df)
+
+# COMMAND ----------
+
+# MAGIC %sql
+# MAGIC CREATE DATABASE IF NOT EXISTS ship2ship
+
+# COMMAND ----------
+
+(AIS_df.write.format("delta").mode("overwrite").saveAsTable("ship2ship.AIS"))
+
+# COMMAND ----------
+
+# MAGIC %md ## Continental US Data
+
+# COMMAND ----------
+
+from pyspark.sql.functions import *
+
+one_metre = 0.00001 - 0.000001
+tolerance = 500 * one_metre
+
+# COMMAND ----------
+
+user_name = (
+    dbutils.notebook.entry_point.getDbutils().notebook().getContext().userName().get()
+)
+
+raw_path = f"dbfs:/tmp/ship2ship/{user_name}"
+raw_us_outline_path = f"{raw_path}/us_outline"
+
+print(f"The raw data is stored in {raw_path}")
+
+# COMMAND ----------
+
+import requests
+import pathlib
+
+us_outline_url = (
+    "https://eric.clst.org/assets/wiki/uploads/Stuff/gz_2010_us_outline_500k.json"
+)
+
+# The DBFS file system is mounted under /dbfs/ directory on Databricks cluster nodes
+
+local_us_outline_path = pathlib.Path(raw_us_outline_path.replace("dbfs:/", "/dbfs/"))
+local_us_outline_path.mkdir(parents=True, exist_ok=True)
+
+req = requests.get(us_outline_url)
+with open(local_us_outline_path / f"us_outline.geojson", "wb") as f:
+    f.write(req.content)
+
+# COMMAND ----------
+
+outlines = (
+    spark.read.format("json")
+    .option("multiline", "true")
+    .load(raw_us_outline_path)
+    .select("type", explode(col("features")).alias("feature"))
+    .select(
+        "type",
+        col("feature.properties").alias("properties"),
+        to_json(col("feature.geometry")).alias("json_geometry"),
+    )
+    .withColumn("geometry", mos.st_aswkt(mos.st_geomfromgeojson("json_geometry")))
+    .withColumn("coastal_area", mos.st_buffer("geometry", lit(tolerance)))
+    .where("properties.type == 'COASTAL'")
+    .drop("json_geometry")
+)
+display(outlines)
```

Comment:
It would be great to persist this as a bronze delta table.
If note persisted, the default behaviour of Spark is keeping it in a single partition, therefore the mosaic_explode defined for `coastal_h3` gets executed by a single slot. So we better either persist `outlines` in a delta table, or add a `.repartition(34)` before the `mosaic_explode`.

Reply:
I have deleted this section since this approach is not used later.